stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/service/sync/PushMessageSyncHandler.kt

```diff
@@ -43,18 +41,10 @@ public class PushMessageSyncHandler(private val service: Service) {
      * registered at your app.
      */
     public fun onNewToken(token: String) {
-        if (ChatClient.isInitialized) {
-            ChatClient.instance().onNewTokenReceived(token)
-        } else {
-            val syncConfig = syncModule.encryptedBackgroundSyncConfigStore.get()
-            syncConfig?.let {
-                val config = it
-                val user = User(id = config.userId)
-                val userToken = config.userToken
-                val client = initClient(service, user, userToken, config.apiKey)
-                client.onNewTokenReceived(token)
-            }
+        syncModule.encryptedBackgroundSyncConfigStore.get()?.let {
```

Comment:
`ChatClient` is always initilized if it is done in the Application class. There are two possible case:

-  App in the background, but the process is alive: `ChatClient` already initilizaed.
-  App process is killed. Then the following happens in this particular order:
1. Content providers are innitialized (irrelevant for us)
2. `Application` class is initialized, and therefore `ChatClient` is initilizaed
3. Services are initialized, and in this particular case `onNewToken` is invoked

Reply:
Not all users are initializing `ChatClient` on `Application`